sunbird-es-utils/src/main/java/org/sunbird/common/ElasticSearchRestHighImpl.java

```diff
@@ -0,0 +1,907 @@
+
+    ActionListener<SearchResponse> listener =
+        new ActionListener<SearchResponse>() {
+          @Override
+          public void onResponse(SearchResponse response) {
+            if (response.getHits() == null || response.getHits().getTotalHits() == 0) {
+              promise.success(new HashMap<>());
+            } else {
+              List<Map<String, Object>> esSource = new ArrayList<>();
+              Map<String, Object> responseMap = new HashMap<>();
+              long count = 0;
+              if (response != null) {
+                SearchHits hits = response.getHits();
+                count = hits.getTotalHits();
+
+                for (SearchHit hit : hits) {
+                  esSource.add(hit.getSourceAsMap());
+                }
+
+                // fetch aggregations aggregations
+                if (null != searchDTO.getFacets() && !searchDTO.getFacets().isEmpty()) {
+                  Map<String, String> m1 = searchDTO.getFacets().get(0);
+                  for (Map.Entry<String, String> entry : m1.entrySet()) {
+                    String field = entry.getKey();
+                    String aggsType = entry.getValue();
+                    List<Object> aggsList = new ArrayList<>();
+                    Map facetMap = new HashMap();
+                    if (JsonKey.DATE_HISTOGRAM.equalsIgnoreCase(aggsType)) {
+                      Histogram agg = response.getAggregations().get(field);
+                      for (Histogram.Bucket ent : agg.getBuckets()) {
+                        // DateTime key = (DateTime) ent.getKey(); // Key
+                        String keyAsString = ent.getKeyAsString(); // Key as String
+                        long docCount = ent.getDocCount(); // Doc count
+                        Map internalMap = new HashMap();
+                        internalMap.put(JsonKey.NAME, keyAsString);
+                        internalMap.put(JsonKey.COUNT, docCount);
+                        aggsList.add(internalMap);
+                      }
+                    } else {
+                      Terms aggs = response.getAggregations().get(field);
+                      for (Bucket bucket : aggs.getBuckets()) {
+                        Map internalMap = new HashMap();
+                        internalMap.put(JsonKey.NAME, bucket.getKey());
+                        internalMap.put(JsonKey.COUNT, bucket.getDocCount());
+                        aggsList.add(internalMap);
+                      }
+                    }
+                    facetMap.put("values", aggsList);
+                    facetMap.put(JsonKey.NAME, field);
+                    finalFacetList.add(facetMap);
+                  }
+                }
+              }
+              responseMap.put(JsonKey.CONTENT, esSource);
+              if (!(finalFacetList.isEmpty())) {
+                responseMap.put(JsonKey.FACETS, finalFacetList);
+              }
+              responseMap.put(JsonKey.COUNT, count);
```

Comment:
use exisitng method to calculate end time and this will be applicable for all places.

Reply:
done